lib/sqlalchemy/dialects/oracle/cx_oracle.py

```diff
@@ -814,6 +821,12 @@ def _returningval(value):
                 self._returningval = self._paramval
 
         self._is_cx_oracle_6 = self.cx_oracle_ver >= (6,)
+        
+        if self.text_encoding_errors and self.cx_oracle_ver < (6, 4):
```

Comment:
not under Python 2, we do the decode ourselves and can still do this.    I'd skip the check here and let it fail at the cx_Oracle level.